src/main/java/de/numcodex/feasibility_gui_backend/query/result/ResultService.java

```diff
@@ -0,0 +1,77 @@
+package de.numcodex.feasibility_gui_backend.query.result;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import de.numcodex.feasibility_gui_backend.query.persistence.ResultType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Synchronized;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class ResultService {
```

Comment:
I would rather add the constructor by hand in service or component classes, because with the contractor in place, IntelliJ will give one the navigation icons for the dependencies. I also would like to add a `final` to the field and use `Objects.requireNonNull` in the constructor.